<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GraphCutSegmentationWidget</class>
 <widget class="QMainWindow" name="GraphCutSegmentationWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>900</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>900</width>
    <height>900</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Interactive Graph Cut Segmentation</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout" rowstretch="1,17,6,0" columnstretch="1,1">
      <item row="0" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Original Image</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Segmented Image</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QVTKWidget" name="qvtkWidgetLeft"/>
      </item>
      <item row="1" column="1">
       <widget class="QVTKWidget" name="qvtkWidgetRight"/>
      </item>
      <item row="3" column="0">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QRadioButton" name="radForeground">
          <property name="text">
           <string>Foreground</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radBackground">
          <property name="text">
           <string>Background</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="1">
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>LambdaMax</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="txtLambdaMax">
            <property name="toolTip">
             <string>The value of Lambda if the slider is set to 100%</string>
            </property>
            <property name="text">
             <string>.001</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>% of LambdaMax</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>0%</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="sldLambda">
            <property name="toolTip">
             <string>The percentage of LambdaMax to use as Lambda</string>
            </property>
            <property name="value">
             <number>50</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>100%</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lblLambda">
            <property name="toolTip">
             <string>The final value of Labmda to use. When Lambda is high, there is no regularization. That is, each pixel decides its label independently (does not care about the labeling of its neighbors).</string>
            </property>
            <property name="text">
             <string>.025</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Histogram Bins Per Dimension:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>1</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="sldHistogramBins">
            <property name="toolTip">
             <string>The higher this value, the more discriminative the regional term will be.</string>
            </property>
            <property name="value">
             <number>40</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>99</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lblHistogramBins">
            <property name="text">
             <string>20</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QPushButton" name="btnCut">
            <property name="text">
             <string>Cut Graph</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QPushButton" name="btnShowStrokesLeft">
          <property name="text">
           <string>Show Strokes</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnHideStrokesLeft">
          <property name="text">
           <string>Hide Strokes</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QPushButton" name="btnShowStrokesRight">
          <property name="text">
           <string>Show Strokes</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnHideStrokesRight">
          <property name="text">
           <string>Hide Strokes</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpenImage"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionFlipImageVertically"/>
    <addaction name="actionFlipImageHorizontally"/>
   </widget>
   <widget class="QMenu" name="menuSelections">
    <property name="title">
     <string>Selections</string>
    </property>
    <addaction name="actionClearForegroundSelection"/>
    <addaction name="actionClearBackgroundSelection"/>
    <addaction name="actionClearAll"/>
    <addaction name="separator"/>
    <addaction name="actionSaveForegroundSelection"/>
    <addaction name="actionSaveBackgroundSelection"/>
    <addaction name="separator"/>
    <addaction name="actionLoadForeground"/>
    <addaction name="actionLoadBackground"/>
   </widget>
   <widget class="QMenu" name="menuExport">
    <property name="title">
     <string>Export</string>
    </property>
    <addaction name="actionExportSegmentedImage"/>
    <addaction name="actionExportSegmentMask"/>
    <addaction name="actionExportScreenshotLeft"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuSelections"/>
   <addaction name="menuExport"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionOpenImage">
   <property name="text">
    <string>Open Image</string>
   </property>
  </action>
  <action name="actionOpen_Grayscale_Image">
   <property name="text">
    <string>Open As Grayscale Image</string>
   </property>
  </action>
  <action name="actionOpen_RGBDI_Image">
   <property name="text">
    <string>Open As RGBDI Image</string>
   </property>
  </action>
  <action name="actionSaveSegmentation">
   <property name="text">
    <string>Save Segmentation</string>
   </property>
  </action>
  <action name="actionFlipImageVertically">
   <property name="text">
    <string>Flip Image Vertically</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionFlipImageHorizontally">
   <property name="text">
    <string>Flip Image Horizontally</string>
   </property>
  </action>
  <action name="actionClearForegroundSelection">
   <property name="text">
    <string>Clear Foreground</string>
   </property>
  </action>
  <action name="actionClearBackgroundSelection">
   <property name="text">
    <string>Clear Background</string>
   </property>
  </action>
  <action name="actionSaveForegroundSelection">
   <property name="text">
    <string>Save Foreground</string>
   </property>
  </action>
  <action name="actionSaveBackgroundSelection">
   <property name="text">
    <string>Save Background</string>
   </property>
  </action>
  <action name="actionExportSegmentedImage">
   <property name="text">
    <string>Segmented Image</string>
   </property>
  </action>
  <action name="actionExportSegmentMask">
   <property name="text">
    <string>Segment Mask</string>
   </property>
  </action>
  <action name="actionLoadForeground">
   <property name="text">
    <string>Load Foreground</string>
   </property>
  </action>
  <action name="actionLoadBackground">
   <property name="text">
    <string>Load Background</string>
   </property>
  </action>
  <action name="actionClearAll">
   <property name="text">
    <string>Clear All</string>
   </property>
  </action>
  <action name="actionExportScreenshotLeft">
   <property name="text">
    <string>Screenshot Left</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QVTKWidget</class>
   <extends>QWidget</extends>
   <header>QVTKWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>sldHistogramBins</sender>
   <signal>valueChanged(int)</signal>
   <receiver>lblHistogramBins</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>786</x>
     <y>765</y>
    </hint>
    <hint type="destinationlabel">
     <x>881</x>
     <y>765</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
